### ISSUE TITLE
Style/SpaceBeforeBlockBraces doesn't run when Style/BlockDelimiters is enabled

### ISSUE THREAD

**Issue by DexterTheDragon** [bbatsov/rubocop#3374]
When using the `Style/SpaceBeforeBlockBraces` cop if the `Style/BlockDelimiters` is also enabled then the `Style/SpaceBeforeBlockBraces` cop is not run. The code ([space_before_block_braces.rb#L15-L22](https://github.com/bbatsov/rubocop/blob/e8a3ebcdd5f2daa6793590e32f9398b56a990740/lib/rubocop/cop/style/space_before_block_braces.rb#L15-L22)) implies that this is intentional saying that if you use braces for a multi-line block then you will get a warning to replace it with do...end in which case the space before the brace doesn't matter. But this is only true when using the `Style/BlockDelimiters` EnforcedStyle setting of `line_count_based`, which prohibits mutli-line blocks using braces.

But if you use the EnforcedStyle setting of either `semantic` or `braces_for_chaining`, both of which allow multi-line blocks to use braces is some cases, then you expect to get a warning from `Style/SpaceBeforeBlockBraces` and don't.

----------

## Steps to reproduce the problem
`test.rb`
```ruby
foo.map{ |a|
  a.bar.to_s
}
```
`test.yml`
```yaml
Style/BlockDelimiters:  
  Enabled: true
  EnforcedStyle: semantic
                                                                                                                                                     
Style/SpaceBeforeBlockBraces:          
  Enabled: true
```

```
$ rubocop --config test.yml --only Style/SpaceBeforeBlockBraces test.rb
```

## Expected behavior
```
Offenses:

test.rb:1:8: C: Space missing to the left of {.
foo.map{ |a|
       ^

1 file inspected, 1 offense detected
```

## Actual behavior
```
1 file inspected, no offenses detected
```
## RuboCop version

```
$ rubocop -V
0.42.0 (using Parser 2.3.1.2, running on ruby 2.3.1 x86_64-linux)
```
